_studio/mfx_lib/scheduler/include/mfx_scheduler_core_thread.h

```diff
@@ -31,25 +34,56 @@ class mfxSchedulerCore;
 
 struct MFX_SCHEDULER_THREAD_CONTEXT
 {
-    MFX_SCHEDULER_THREAD_CONTEXT()
+    typedef std::function<void(MFX_SCHEDULER_THREAD_CONTEXT*)> SchedulerThreadFunc;
+
+    MFX_SCHEDULER_THREAD_CONTEXT() = delete;
+
+    MFX_SCHEDULER_THREAD_CONTEXT(mfxU32 id)
       : state(State::Waiting)
-      , pSchedulerCore(NULL)
-      , threadNum(0)
-      , threadHandle()
+      , threadNum(id)
+      , threadHandle([this]() {
+          start.get_future().wait();
+          if (threadFunc) threadFunc(this);
+      })
       , workTime(0)
       , sleepTime(0)
-    {}
+    {
+    }
+
+    void Start(SchedulerThreadFunc func)
```

Comment:
tricky thing about should it be reference or not. Potentially copying of std::function may cause deep copy involving heap allocation (but as I see you are using lambdas with small capture to pass here, so some internal optimization for small objects may be used which result in cheap copy).

Reply:
I will follow your guidance:).... Since you are not sure I will remain it as is for now.

Reply:
yes, let's remain